services/alertmanager/alertmanager.go

```diff
@@ -181,6 +208,71 @@ func (svc *Service) updateConfiguration(ctx context.Context) {
 	svc.l.Infof("%s created", alertmanagerConfigPath)
 }
 
+func (svc *Service) collectRuleTemplates() {
+	rules := make([]saas.Rule, 0)
+
+	shippedFilePaths, err := getRuleTemplateFilePaths(svc.shippedRuleTemplatePath)
+	if err != nil {
+		svc.l.Errorf("Failed to get paths of template files shipped with PMM: %s.", err)
+		return // keep previously loaded rules
+	}
+
+	for _, path := range shippedFilePaths {
+		r, err := svc.loadRuleTemplates(path)
+		if err != nil {
+			svc.l.Errorf("Failed to load shipped rule template file: %s.", err)
+			return // keep previously loaded rules
+		}
+		rules = append(rules, r...)
+	}
+
+	userDefinedFilePaths, err := getRuleTemplateFilePaths(svc.userDefinedRuleTemplatePath)
+	if err != nil {
+		svc.l.Errorf("Failed to get paths of user-defined template files: %s.", err)
+		return // keep previously loaded rules
+	}
+
+	for _, path := range userDefinedFilePaths {
+		r, err := svc.loadRuleTemplates(path)
+		if err != nil {
+			svc.l.Errorf("Failed to load user-defined rule template file: %s.", err)
+			return // keep previously loaded rules
+		}
+		rules = append(rules, r...)
+	}
+
+	// TODO add method to download templates from SAAS.
+
+	svc.rules = rules
+}
+
+func getRuleTemplateFilePaths(pattern string) ([]string, error) {
+	paths, err := filepath.Glob(pattern)
+	if err != nil {
+		return nil, err
+	}
+	return paths, nil
+}
```

Comment:
Why not use `filepath.Glob` directly? That wrapper doesn't add any benefits as I see, no?

Reply:
because I call the wrapper twice

Reply:
I mean, instead of this:
```
	shippedFilePaths, err := getRuleTemplateFilePaths(svc.shippedRuleTemplatePath)
	if err != nil {
		svc.l.Errorf("Failed to get paths of template files shipped with PMM: %s.", err)
		return // keep previously loaded rules
	}
```

you can write this:
```
	shippedFilePaths, err := filepath.Glob(svc.shippedRuleTemplatePath)
	if err != nil {
		svc.l.Errorf("Failed to get paths of template files shipped with PMM: %s.", err)
		return // keep previously loaded rules
	}
```
and there will be no difference.